### DISCUSSION TITLE
Startup probes failed on new CephObjectStore

### DISCUSSION THREAD

**Discussion by barside-code** [rook/rook#10292] [Q&A]
Hi,
I'm trying to create my first CephObjectStore with S3 access based on documentation https://rook.io/docs/rook/v1.9/ceph-object.html. When I create it, the relative pod marked as **rook-ceph-rwg-<object-name>** failed to startup because on a startup probes failed.

This is my pod describe

  Normal   Scheduled  49m                    default-scheduler  Successfully assigned rook-ceph/rook-ceph-rgw-<object-name>-a-75cbd99d5-rjpg7 to aks-agentpool-ZZZZZZZZZ
  Normal   Pulled     49m                    kubelet            Container image "quay.io/ceph/ceph:v16.2.7" already present on machine
  Normal   Created    49m                    kubelet            Created container chown-container-data-dir
  Normal   Started    49m                    kubelet            Started container chown-container-data-dir
  Normal   Pulled     49m                    kubelet            Container image "quay.io/ceph/ceph:v16.2.7" already present on machine
  Normal   Created    49m                    kubelet            Created container rgw
  Normal   Started    49m                    kubelet            Started container rgw
  Normal   Killing    46m                    kubelet            Container rgw failed startup probe, will be restarted
  Warning  Unhealthy  9m39s (x174 over 49m)  kubelet            Startup probe failed: dial tcp 10.1.66.102:8080: connect: connection refused
  Warning  BackOff    4m38s (x42 over 20m)   kubelet            Back-off restarting failed container

And this is my yaml

apiVersion: ceph.rook.io/v1
kind: CephObjectStore
metadata:
  name: portale-aste-osd
  namespace: rook-ceph
spec:
  metadataPool:
    failureDomain: host
    replicated:
      size: 3
  dataPool:
    failureDomain: host
    erasureCoded:
      dataChunks: 2
      codingChunks: 1
  preservePoolsOnDelete: true
  gateway:
    port: 80
    instances: 1
  healthCheck:
    bucket:
      disabled: false
      interval: 60s

Can anyone help me please?

**Comment by thotz** [comment]
Can you please close the https://github.com/rook/rook/issues/10293 and will have discussion here

Please provide more information to debug like 
* operator pod logs (preferably in debug)
* `ceph status` from toolbox pod
* kubectl get pods -n rook-ceph
* kuebctl -n rook-ceph describe cephobjectstores

**Reply by barside-code** [reply]
Hi these are the informations:

- operator pod logs -> attached file "operator.log";
- ceph status:
 `  cluster:
    id:     3a22f787-df32-47d1-83fb-5714d314d097
    health: HEALTH_WARN
            1 MDSs report slow metadata IOs
            Reduced data availability: 197 pgs inactive, 64 pgs incomplete
            Degraded data redundancy: 167 pgs undersized

  services:
    mon: 3 daemons, quorum a,b,c (age 2d)
    mgr: b(active, since 2d), standbys: a
    mds: 1/1 daemons up, 1 standby
    osd: 3 osds: 3 up (since 2d), 3 in (since 2d); 44 remapped pgs

  data:
    volumes: 1/1 healthy
    pools:   16 pools, 241 pgs
    objects: 13 objects, 860 B
    usage:   50 MiB used, 300 GiB / 300 GiB avail
    pgs:     81.743% pgs not active
             16/39 objects misplaced (41.026%)
             133 undersized+peered
             64  creating+incomplete
             34  active+undersized+remapped
             10  active+clean

  progress:
    Global Recovery Event (0s)
      [............................]`

- kubectl get pods -n rook-ceph
`NAME                                                              READY   STATUS             RESTARTS        AGE
csi-cephfsplugin-2x4f5                                            3/3     Running            0               2d23h
csi-cephfsplugin-bmjxr                                            3/3     Running            0               2d23h
csi-cephfsplugin-h9knq                                            3/3     Running            0               2d23h
csi-cephfsplugin-hrd2m                                            3/3     Running            0               2d23h
csi-cephfsplugin-jzvcp                                            3/3     Running            0               2d23h
csi-cephfsplugin-provisioner-6ffb5fd4d7-rs6fk                     6/6     Running            7 (10h ago)     2d23h
csi-cephfsplugin-provisioner-6ffb5fd4d7-wgnmc                     6/6     Running            8 (2d19h ago)   2d23h
csi-cephfsplugin-ss4fm                                            3/3     Running            0               2d23h
csi-cephfsplugin-tcklw                                            3/3     Running            0               2d23h
csi-rbdplugin-6tcbz                                               3/3     Running            0               2d23h
csi-rbdplugin-h2ktl                                               3/3     Running            0               2d23h
csi-rbdplugin-jq8vs                                               3/3     Running            0               2d23h
csi-rbdplugin-mjkvl                                               3/3     Running            0               2d23h
csi-rbdplugin-provisioner-874767cb7-288gr                         6/6     Running            8 (10h ago)     2d23h
csi-rbdplugin-provisioner-874767cb7-669ss                         6/6     Running            8 (2d19h ago)   2d23h
csi-rbdplugin-q4mgq                                               3/3     Running            0               2d23h
csi-rbdplugin-vm5x2                                               3/3     Running            0               2d23h
csi-rbdplugin-wz9rq                                               3/3     Running            0               2d23h
rook-ceph-crashcollector-aks-agentpool-41322224-vmss000000dc2qz   1/1     Running            0               2d23h
rook-ceph-crashcollector-aks-agentpool-41322224-vmss0000012dhv8   1/1     Running            0               2d23h
rook-ceph-crashcollector-aks-agentpool-41322224-vmss000002k6zd6   1/1     Running            0               2d23h
rook-ceph-crashcollector-aks-agentpool-41322224-vmss000003nldnh   1/1     Running            0               2d23h
rook-ceph-crashcollector-aks-npceph-22073080-vmss000001-d544p55   1/1     Running            0               2d23h
rook-ceph-mds-ceph-filesystem-a-687c498488-4h2hh                  1/1     Running            0               2d23h
rook-ceph-mds-ceph-filesystem-b-6f676f8898-294nm                  1/1     Running            0               2d23h
rook-ceph-mgr-a-96c95d6f5-7xk4g                                   2/2     Running            0               2d23h
rook-ceph-mgr-b-687b564cc7-hhqbd                                  2/2     Running            0               2d23h
rook-ceph-mon-a-6c6bb577c9-vdxkm                                  1/1     Running            0               2d23h
rook-ceph-mon-b-dcd4f4bf6-nf4dl                                   1/1     Running            0               2d23h
rook-ceph-mon-c-7cc4cf4f6f-j7nq8                                  1/1     Running            0               2d23h
rook-ceph-operator-56884d57df-8smsk                               1/1     Running            0               3d
rook-ceph-osd-0-858bd89f8f-5x5vj                                  1/1     Running            0               2d23h
rook-ceph-osd-1-75cdc8968f-h52tf                                  1/1     Running            0               2d23h
rook-ceph-osd-2-57bd9db7c5-brzt5                                  1/1     Running            0               2d23h
rook-ceph-osd-prepare-7292d9d073a72a18d8236e7e15ab8abe--1-pk5r2   0/1     Completed          0               2d23h
rook-ceph-osd-prepare-b9d84cd8e340b3a7df79fd83542ca34e--1-2xnqw   0/1     Completed          0               2d23h
rook-ceph-osd-prepare-bc5d006b065eacbe876f8b11c6086630--1-bmdml   0/1     Completed          0               2d23h
rook-ceph-osd-prepare-cc94d0a1ee734fc66347f7e6443c89cb--1-8npqh   0/1     Completed          0               2d23h
rook-ceph-osd-prepare-set1-data-0g997p--1-4fppv                   0/1     Completed          0               2d23h
rook-ceph-osd-prepare-set1-data-1grk6f--1-tljht                   0/1     Completed          0               2d23h
rook-ceph-osd-prepare-set1-data-2kgqmf--1-h69cl                   0/1     Completed          0               2d23h
rook-ceph-rgw-portale-aste-osd-a-75cbd99d5-rjpg7                  0/1     Running            696 (93s ago)   2d23h
rook-ceph-rgw-portale-aste-store-a-6f5c7bfbd-ljzbt                0/1     CrashLoopBackOff   596 (63s ago)   2d19h
rook-ceph-tools-54b94d5b76-zphgs                                  1/1     Running            0               2d23h`

- kuebctl -n rook-ceph describe cephobjectstores -> attached file "cephObjectStores.log"

Regards,

[cephObjectStores.log](https://github.com/rook/rook/files/8752695/cephObjectStores.log)
[operator.log](https://github.com/rook/rook/files/8752712/operator.log)

**Reply by barside-code** [reply]
Hi,
sorry, I'm attaching also the pods list
![pods](https://user-images.githubusercontent.com/3807303/169775187-d2bd5f1e-683b-4339-a4b6-2e78de2f4355.png)

**Comment by thotz** [comment]
I can see following logs consistently in operator logs
```
clusterdisruption-controller: all "host" failure domains: [aks-npceph-22073080-vmss000001]. osd is down in failure domain: "". active node drains: false. pg health: "cluster is not fully clean. PGs: [{StateName:undersized+peered Count:133} {StateName:creating+incomplete Count:64} {StateName:active+undersized+remapped Count:34} {StateName:active+clean Count:10}]"
```

I am guessing PGs are not available for performing RGW operations hence it is not coming up

@travisn : any ideas on how to recover???

**Reply by travisn** [reply]
The key looks like the PGs are not clean:
```
pgs: 81.743% pgs not active
16/39 objects misplaced (41.026%)
133 undersized+peered
64 creating+incomplete
34 active+undersized+remapped
10 active+clean
```

Since the OSDs are all up and in, it's not obvious why ceph is not bringing the PGs back online. A few questions:
- Is there anything interseting in the osd or mon logs? 
- What is the `failureDomain` for the pool(s)?
- What is the OSD topology? (`ceph osd tree`)
